skipchain/skipchain.go

```diff
@@ -270,12 +261,10 @@ func (s *Service) GetSingleBlockByIndex(id *GetSingleBlockByIndex) (*SkipBlock,
 // GetAllSkipchains returns a list of all known skipchains
 func (s *Service) GetAllSkipchains(id *GetAllSkipchains) (*GetAllSkipchainsReply, onet.ClientError) {
 	// Write all known skipblocks to a map, thus removing double blocks.
-	chains := map[string]*SkipBlock{}
-	s.Sbm.Lock()
-	for _, sb := range s.Sbm.SkipBlocks {
-		chains[string(sb.SkipChainID())] = sb
+	chains, err := s.db.dump()
```

Comment:
I am not a super-fan of such a simple, overused word as "dump" for a method name. Maybe "all()"? Especially if later we need a filtering thing called select(func f(...) bool).

Reply:
`all` like a high order function, e.g. `all(func (k Key, v Value) bool) (map[Key]Value, error)`?

Reply:
Actually, what I described is usually called filter. All returns a bool if all items satisfies the predicate.